### ISSUE TITLE
Input focus not properly switched between screens

### ISSUE THREAD

**Issue by UltraBlackLinux** [WayfireWM/wayfire#1636]
**Describe the bug**
Clicking on a window on another screen does not properly switch the input too. Oftentimes I'm left with a focused text field which cannot be typed into, while those keystrokes are sent to the previously focused window.

As far as I remember this started to happen when I updated my system yesterday

**Wayfire version**
 r2369.c3b1edf1

**Comment by UltraBlackLinux** [comment]
I just figured out how to reproduce it reliably:
1. Focus a window on screen 1
2. Move that window to screen 2
3. Focus a window on screen 1
4. Type into that window
5. tries to type into the window on screen 1

**Comment by ammen99** [comment]
Does #1640 fix the issue?

**Comment by UltraBlackLinux** [comment]
> Does #1640 fix the issue?

Just tried, no. Sadly still misbehaving. The inputs still don't end up in the correct window.

**Comment by bklaase** [comment]
As I mention in the closing comments of #1633, it seems to matter how you focus a window (clicking, workspace / monitor switching, or by plugin (a switcher) )

It can also help in dissecting to have a bar highlighting the active window. Since the behavior seems to have layers:
- focused workspace / mon
- -focused window
- input focus

**Comment by UltraBlackLinux** [comment]
I just usually click between windows. No switchers consider windows on other screens.

> It can also help in dissecting to have a bar highlighting the active window. Since the behavior seems to have layers:
Yeah I already have that set up.

While we are at it - I noticed something with my active windows: Moving an active window from screen 1 to screen two results in two different active windows on two different screens (indicated by the borders at least) - This might cause the issue

**Comment by ammen99** [comment]
Oh, this isssue is about multiple outputs, not multiple workspaces ...

**Comment by UltraBlackLinux** [comment]
> Oh, this isssue is about multiple outputs, not multiple workspaces ...

yeah, that's why I was talking about Screens in the first place. I don't wanna mix that up with anything else ;)

**Comment by UltraBlackLinux** [comment]
Still clueless? This issue is driving me crazy since I manage to trigger it pretty much every second window switch. I hope this can be fixed easily...

**Comment by bklaase** [comment]
Are you on latest commit?

Can you check whether I am trying to reproduce correctly:

What I do, is I have external monitor (screen 1) and internal monitor (screen 2)
I open two alacritty windows (wayland clients, not xwayland) on screen 1.

Then per your recipe:

1. Focus window A on screen 1 - starting state
2. Move window A to screen 2 - done using `osswitch->next_output_with_win`
3. Focus window B on screen 1 - done by moving mouse back to screen 1, and clicking window B
4.Type into window B - works as expected
5.Tries to type into window A - does not happen

If you can correct the way I execute one of the steps. Perhaps you use xwayland clients? perhaps different ways of switching windows / screens?

**Comment by UltraBlackLinux** [comment]
Just checked again, and it does appear to be an issue with xwayland, which I didn't consider at the time. I tried to reproduce this on kitty, which I did not succeed with.

I also want to mention that I'm moving my windows with the mouse using the move plugin.
